modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -691,22 +705,51 @@ def log_direct_schedule_submission_errors(e)
       end
 
       def set_type(appointment)
-        type = APPOINTMENT_TYPES[:request] if appointment[:kind] != 'cc' && appointment[:request_periods].present?
-
-        type ||= case appointment[:kind]
-                 when 'cc'
-                   if appointment[:start]
-                     APPOINTMENT_TYPES[:cc_appointment]
-                   else
-                     APPOINTMENT_TYPES[:cc_request]
-                   end
-                 else
-                   APPOINTMENT_TYPES[:va]
-                 end
+        type = if cerner?(appointment)
+                 cerner_type(appointment)
+               else
+                 non_cerner_type(appointment)
+               end
 
         appointment[:type] = type
       end
 
+      # Determines the type of appointment for Cerner appointments.
+      # @param appointment [Hash] the appointment to determine the type for
+      #
+      # @return [String] the type of appointment
+      #
+      def cerner_type(appointment)
+        if appointment[:end].present?
+          APPOINTMENT_TYPES[:va]
+        elsif appointment[:requested_periods].present?
+          appointment[:kind] == 'cc' ? APPOINTMENT_TYPES[:cc_request] : APPOINTMENT_TYPES[:request]
+        else
+          APPOINTMENT_TYPES[:request]
+        end
+      end
+
+      # Determines the type of appointment for non-Cerner appointments.
+      # @param appointment [Hash] the appointment to determine the type for
+      #
+      # @return [String] the type of appointment
+      #
+      def non_cerner_type(appointment)
+        if appointment[:kind] == 'cc'
+          if appointment[:start].present?
+            APPOINTMENT_TYPES[:cc_appointment]
+          elsif appointment[:requested_periods].present?
+            APPOINTMENT_TYPES[:cc_request]
+          else
+            APPOINTMENT_TYPES[:va]
```

Comment:
If the appointment kind is "cc" I don't think we should ever set the type to "VA".

Reply:
This was my attempt to implement the logic that currently resides in the FE [here](https://github.com/department-of-veterans-affairs/vets-website/blob/20a2e97a6b46ec4cd4d8487db3157063f5644632/src/applications/vaos/services/appointment/transformers.js#L13).  

For the corner case where a community case appointment has _neither_ a start date or request period, we are currently setting the type to VA appointment.  I'm happy to change this if needed, @simiadebowale it may make more sense to default the type to cc_appointment for community care appointments.

Reply:
I agree, I don't think we will ever want to default to setting a CC appointment to 'VA'

Reply:
Thanks @simiadebowale and @JunTaoLuo, I've updated the logic to set the type to CC appointment/request if the type is 'cc'.